### PR TITLE
Plugin implementation

### PR DESCRIPTION
**Pull request by avkos**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "web3-plugin-template",
+  "name": "web3-plugin-tx-eip4844",
```
Comment (jdevcs) [inline package.json]: this will need rename to `web3-plugin-blob-tx` accordign to new repo

File: src/index.ts [hunk 2]
```diff
@@ -1,16 +1,24 @@
+import { TransactionFactory } from "web3-eth-accounts";
 import { Web3PluginBase } from "web3";
+import { BlobEIP4844Transaction, TransactionType } from "./eip4844";
 
-export class TemplatePlugin extends Web3PluginBase {
-  public pluginNamespace = "template";
+export class Web3TxEIP4844Plugin extends Web3PluginBase {
+  public pluginNamespace = "txTypeEIP4844Plugin";
```
Comment (jdevcs) [inline src/index.ts]: rename here as well

Comment (jdevcs) [reply src/index.ts]: and other places in other files

File: package.json [hunk 3]
```diff
@@ -19,23 +19,30 @@
   "license": "MIT",
   "repository": {
     "type": "git",
-    "url": "git@github.com:web3/web3.js-plugin-template.git"
+    "url": "git@github.com:web3/web3.js-plugin-tx-eip4844.git"
```
Comment (jdevcs) [inline package.json:23]: update required for : https://github.com/web3/web3.js-plugin-blob-tx

File: src/index.ts [hunk 4]
```diff
@@ -1,16 +1,54 @@
...
+  EIP4844,
+  hardforks,
+  CHAINS,
+  transactionSchemaEip4844,
+  EIP4895,
+} from "./shemas";
 
-export class TemplatePlugin extends Web3PluginBase {
-  public pluginNamespace = "template";
+export class Web3TxEIP4844Plugin extends Web3PluginBase {
```
Comment (jdevcs) [inline src/index.ts]:
```suggestion
export class Web3BlobTxPlugin extends Web3PluginBase {
```

File: test/index.test.ts [hunk 5]
```diff
@@ -1,31 +1,180 @@
-import Web3, { core } from "web3";
-import { TemplatePlugin } from "../src";
+import { Web3 } from "web3";
+import { randomBytes } from "@ethereumjs/util";
+import { Web3TxEIP4844Plugin } from "../src";
+import { concatBytes } from "./testData/util";
 
-describe("TemplatePlugin Tests", () => {
```
Comment (jdevcs) [inline test/index.test.ts:4]: could you also add local E2E test for sepolia NW for sending blob tx test with PK in env var

**Comment by avkos** [comment]
Build will not work until we not merge and make new realse with [these changes](https://github.com/web3/web3.js/pull/7000)

After this I need to change version of web3

File: src/plugin.ts [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+								tx: transactionFormatted as TransactionCall,
+							}),
+						);
+					}
+				});
+			},
+		);
+		return promiEvent;
+	}
+	async getTransactionReceipt<ReturnFormat extends DataFormat>(
```
Comment (jdevcs) [inline src/plugin.ts:290]: support of `getTransaction` and `getTransactionFromBlock` for blob tx is also required

**Review by jdevcs — COMMENTED** [review]
lgtm, except after merging PR in main web3.js tests should be passing for merging this PR. Thanks @avkos

File: src/plugin.ts [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+
+		const common = new Common({
+			chain: chain ?? this.defaultChain,
+			hardfork: hardfork ?? this.defaultHardfork,
+			eips: [4844],
+			customCrypto: customCrypto ?? this.crypto,
+		});
+		return BlobEIP4844Transaction.fromTxData(
+			{
+				type: 5,
```
Comment (jdevcs) [inline src/plugin.ts]: this should be 0x3 https://eips.ethereum.org/EIPS/eip-4844

**Review by jdevcs — APPROVED** [review]

**Pull request merged**